pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -140,6 +147,19 @@ func (h *VMController) UnsetOwnerOfDataVolumes(_ string, vm *kv1.VirtualMachine)
 			return vm, fmt.Errorf("failed to revoke VitrualMachine(%s/%s) as DataVolume(%s/%s)'s owner: %w",
 				vm.Namespace, vm.Name, dataVolumeNamespace, dataVolumeName, err)
 		}
+
+		if isExistingVolumeToDelete {
+			numberOfOwner, err := numberOfBoundedDataVolumeReference(h.dataVolumeClient, dataVolume, vm)
+			if err != nil {
+				return vm, fmt.Errorf("failed to count number of owners for DataVolume(%s/%s): %w", dataVolumeNamespace, dataVolumeName, err)
+			}
+			// We are the solely owner here, so try to delete the data volume as requested.
+			if numberOfOwner == 1 {
+				if err := h.dataVolumeClient.Delete(dataVolumeNamespace, dataVolumeName, &metav1.DeleteOptions{}); err != nil {
+					return vm, err
+				}
+			}
```

Comment:
What if numberOfOwner>1, is that a valid use case? Seems like it is simply ignored and proceeds but the volume is not deleted.

Reply:
At this moment, a data volume can be owned by multiple virtual machines (not sure whether that is expected). To deal with that situation, we only delete data volume meets these criteria:

- has `RemovedDataVolumesAnnotationKey` annotation
- has a solely owner which is also the mv in the scope of `UnsetOwnerOfDataVolumes`

Therefore, I think we can safely ignore cases with more than one owner.